cmake/kokkos_arch.cmake

```diff
@@ -480,24 +480,19 @@ IF (KOKKOS_ARCH_NATIVE AND NOT "${KOKKOS_COMPILE_OPTIONS}" STREQUAL "${KOKKOS_CO
   SET(CMAKE_REQUIRED_FLAGS "${KOKKOS_COMPILE_OPTIONS}")
   INCLUDE(CheckCXXSymbolExists)
 
-  # Make sure to rerun the check if compile options have changed
+  # Make sure to rerun the checks if compile options have changed
   UNSET(KOKKOS_COMPILER_HAS_AVX512 CACHE)
   CHECK_CXX_SYMBOL_EXISTS(__AVX512F__ "" KOKKOS_COMPILER_HAS_AVX512)
-  IF(KOKKOS_COMPILER_HAS_AVX512)
-    SET(KOKKOS_ARCH_AVX512XEON ON)
-  ENDIF()
-
-  # Make sure to rerun the check if compile options have changed
   UNSET(KOKKOS_COMPILER_HAS_AVX2 CACHE)
   CHECK_CXX_SYMBOL_EXISTS(__AVX2__ "" KOKKOS_COMPILER_HAS_AVX2)
-  IF(KOKKOS_COMPILER_HAS_AVX2)
-    SET(KOKKOS_ARCH_AVX2 ON)
-  ENDIF()
-
-  # Make sure to rerun the check if compile options have changed
   UNSET(KOKKOS_COMPILER_HAS_AVX CACHE)
   CHECK_CXX_SYMBOL_EXISTS(__AVX__ "" KOKKOS_COMPILER_HAS_AVX)
-  IF(KOKKOS_COMPILER_HAS_AVX)
+
+  IF(KOKKOS_COMPILER_HAS_AVX512)
```

Comment:
I suggest leaving a comment here that briefly notes on these macros needing to be defined mutually exclusively for now and that this will probably have to be looked at again in the future.